app/src/main/java/com/helio/app/networking/ScheduleDeserializer.java

```diff
@@ -0,0 +1,38 @@
+package com.helio.app.networking;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+import com.helio.app.model.Day;
+import com.helio.app.model.Schedule;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This custom deserializer is needed because the JSON contains the day as a String, but the app stores it as a type {@link Day}.
+ */
+public class ScheduleDeserializer implements JsonDeserializer<Schedule> {
+
+    @Override
+    public Schedule deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+        Gson gson = new Gson();
+        Schedule schedule = gson.fromJson(json, Schedule.class);
+
+        // Get the element from the json
+        JsonObject jsonObject = json.getAsJsonObject();
+        JsonElement jsonElement = jsonObject.get("days");
+
+        // Convert to type Day
+        List<String> dayStrings = gson.fromJson(jsonElement, new TypeToken<List<String>>() {}.getType());
+        schedule.setDays(dayStrings.stream().map(Day::getEnumFromName).collect(Collectors.toList()));
```

Comment:
I'm being warned about using these newer features without explicitly requiring API level 24:
![Screenshot from 2021-02-23 16-58-15](https://user-images.githubusercontent.com/30589554/108879705-6d8a0380-75f9-11eb-8437-baa0f1219234.png)

Reply:
I got the same until I added desugaring to the gradle (https://developer.android.com/studio/write/java8-support#library-desugaring). I tried removing it from the gradle and immediately got the error again, then put it back and it went away. Maybe you have to re-sync the gradle, or restart it, or something like that to get it to work?